# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.h
# Commit: f441cb5e7e6e
# Full Hash: f441cb5e7e6e30d4a60b08ca99e56a29d3dc3a8b
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r cfae5efaae37 -r f441cb5e7e6e dom/script/ScriptSettings.h
--- a/dom/script/ScriptSettings.h	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/script/ScriptSettings.h	Mon Mar 31 19:49:02 2025 +0000
@@ -35,6 +35,7 @@
 namespace dom {
 
 class Document;
+class WebTaskSchedulingState;
 
 /*
  * Per thread setup/teardown routines. Init and Destroy should be invoked
@@ -95,6 +96,8 @@
 // Returns the global associated with the current compartment. This may be null.
 nsIGlobalObject* GetCurrentGlobal();
 
+WebTaskSchedulingState* GetWebTaskSchedulingState();
+
 // JS-implemented WebIDL presents an interesting situation with respect to the
 // subject principal. A regular C++-implemented API can simply examine the
 // compartment of the most-recently-executed script, and use that to infer the